modular_skyrat/modules/modular_implants/code/nifs.dm

```diff
@@ -0,0 +1,511 @@
+		return FALSE
+
+	if((length(loaded_nifsofts) >= max_nifsofts))
+		return FALSE
+
+	if(!is_type_in_list(src, loaded_nifsoft.compatible_nifs))
+		send_message("[loaded_nifsoft.name] is incompatible with your NIF!", TRUE)
+		return FALSE
+
+	for(var/datum/nifsoft/current_nifsoft as anything in loaded_nifsofts)
+		if(loaded_nifsoft.single_install && loaded_nifsoft.type == current_nifsoft.type)
+			send_message("Multiple of [loaded_nifsoft.name] cannot be installed", TRUE)
+			return FALSE
+
+		if(current_nifsoft.type in loaded_nifsoft.mutually_exclusive_programs)
+			send_message("[current_nifsoft.name] is preventing [loaded_nifsoft.name] from being installed", TRUE)
+			return FALSE
+
+	loaded_nifsofts += loaded_nifsoft
+	loaded_nifsoft.parent_nif = src
+	loaded_nifsoft.linked_mob = linked_mob
+
+	send_message("[loaded_nifsoft.name] has been added")
+	return TRUE
+
+/obj/item/organ/internal/cyberimp/brain/nif/proc/remove_nifsoft(datum/nifsoft/removed_nifsoft, silent = FALSE)
+	if(!is_type_in_list(removed_nifsoft, loaded_nifsofts) || broken)
+		return FALSE
+
+	if(!silent)
+		send_message("[removed_nifsoft.name] has been removed", TRUE)
+
+	qdel(removed_nifsoft)
+	return TRUE
+
+/obj/item/organ/internal/cyberimp/brain/nif/proc/repair_nif(repair_amount)
+	if(durability == max_durability)
+		return FALSE
+
+	durability += repair_amount
+	if(durability > max_durability)
+		durability = max_durability
+
+	return TRUE
+
+/obj/item/organ/internal/cyberimp/brain/nif/proc/send_message(message_to_send, alert=FALSE)
+	var/datum/asset/spritesheet/sheet = get_asset_datum(/datum/asset/spritesheet/chat)
+	var/tag = sheet.icon_tag("nif-[chat_icon]")
+	var/nif_icon = ""
+
+	if(tag)
+		nif_icon = tag
+
+	if(alert)
+		to_chat(linked_mob, span_warning("[nif_icon] <b>NIF Alert</b>: [message_to_send]"))
+		linked_mob.playsound_local(linked_mob, bad_sound, 60, FALSE)
+		return
+
+	to_chat(linked_mob, span_cyan("[nif_icon] <b>NIF Message</b>: [message_to_send]"))
+	linked_mob.playsound_local(linked_mob, good_sound, 60, FALSE)
+
+
+///Fixes the NIF if it is broken. This doesn't not repair durability
+/obj/item/organ/internal/cyberimp/brain/nif/proc/fix_nif()
+	if(!broken)
+		return FALSE
+
+	broken = FALSE
+	send_message("Your NIF is now in working condition!")
+
+///Re-enables the durability_loss_vulnerable variable
+/obj/item/organ/internal/cyberimp/brain/nif/proc/make_vulnerable()
+	durability_loss_vulnerable = TRUE
+
+/obj/item/organ/internal/cyberimp/brain/nif/attack_self(mob/user, modifiers)
+	return FALSE
+
+/obj/item/organ/internal/cyberimp/brain/nif/emp_act(severity)
+	. = ..()
+
+	if(!durability_loss_vulnerable)
+		return FALSE
+
+	broken = TRUE
+	durability_loss_vulnerable = FALSE
+
+	addtimer(CALLBACK(src, .proc/fix_nif), 30 SECONDS)
+	addtimer(CALLBACK(src, .proc/make_vulnerable), 3 MINUTES)
+
+	switch(severity)
+		if(1)
+			durability -= death_durability_loss
+		if(2)
+			durability -= (death_durability_loss / 2)
```

Comment:
Defines please!

Reply:
Please elaborate on what you want as a define, other instances I have seen for EMP code do not use DEFINES to switch between states.